common/lib/common-definitions/src/logger.ts

```diff
@@ -29,11 +29,32 @@ export interface ITaggedTelemetryPropertyType {
 
 /**
  * JSON-serializable properties, which will be logged with telemetry.
+ * @deprecated interface will be repurposed to support flat arrays and objects.
  */
 export interface ITelemetryProperties {
     [index: string]: TelemetryEventPropertyType | ITaggedTelemetryPropertyType;
 }
 
+/**
+ * Set of properties defined for the base logger to use only primitive types.
+ */
+// eslint-disable-next-line @rushstack/no-new-null
+export type TelemetryBaseEventPropertyType = string | number | boolean | null | undefined;
+
+/**
+ * A property to be logged to telemetry containing both the value and a tag. Tags are generic strings that can be used
+ * to mark pieces of information that should be organized or handled differently by loggers in various first or third
+ * party scenarios. For example, tags are used to mark PII that should not be stored in logs.
+ */
+export interface ITaggedBaseTelemetryPropertyType {
```

Comment:
While we're here introducing new names, I would call these `TelemetryBaseEventPropertyValue` and `ITaggedBaseTelemetryProperty`.  Maybe we can rename the existing ones eventually too as we repurpose them.